File: cpp/Problem27.cpp
/*
 * Problem: 27. Remove Element
 * URL: https://leetcode.com/problems/remove-element/
 * Difficulty: Easy
 * Tags: Array
 */

#include <vector>

using namespace std;

class Solution {
public:
	int removeElement(vector<int>& nums, int val) {
		int numSize = 0, size = nums.size();
		for (int i = 0; i < size; i++) {
			if (nums[i] != val) {
				nums[numSize] = nums[i];
				numSize++;
			}
		}
		return numSize;
	}
};

File: python/Problem27.py
# Problem: 27. Remove Element
# URL: https://leetcode.com/problems/remove-element/
# Difficulty: Easy
# Tags: Array

from typing import List

class Solution:
  def removeElement(self, nums: List[int], val: int) -> int:
    newLength: int = 0
    length: int = len(nums)
    for i in range(length):
      if nums[i] != val:
        nums[newLength] = nums[i]
        newLength += 1
    return newLength